labs/src/main/scala/org/scalalabs/basic/lab03/PatternMatchingExercise.scala

```diff
@@ -47,7 +53,15 @@ object PatternMatchingExercise {
    *    - anything else, the function result is "Some Scala class"
    */
   def matchOnInputType(in: Any): String = {
-    error("fix me")
+    in match {
+      case str: String => s"A string with length ${str.length}"
+      case n: Integer if n > 0 => "A positive integer"
+      case person: Person => s"A person with name: ${person.name}"
+      case seq: Seq[_] if seq.size > 10 => "Seq with more than 10 elements"
```

Comment:
This was my first solution too, but it doesn’t work on infinite sequences like `Stream continually 1`. You can tell relatively quickly that those are larger than 10, so they should pattern match and return `"Seq with more than 10 elements"`; however, they instead get caught on infinite recursion with the `size` method.